### DISCUSSION TITLE
Build issue with the peer versioning?

### DISCUSSION THREAD

**Discussion by TommyLeong** [invertase/react-native-firebase#4583] [Q&A] [answered]
I'm trying to build a Production (release) app (Android-apk), however I'm facing the issue, the issue was however not appearing when I build for my Staging/UAT (release) app. 

Here's my RNFirebase dependency, yes I'm aware some version are quite old. As of today (21 Nov 2020), I saw version `10.0.0` is available at NPM.
```
"@react-native-firebase/analytics": "^7.3.1",
    "@react-native-firebase/app": "^8.3.0",
    "@react-native-firebase/crashlytics": "^8.1.2",
    "@react-native-firebase/messaging": "^7.9.2",
```

**Issue:**
```
ReactNativeFirebase WARNING: NPM package '@react-native-firebase/messaging' depends on '@react-native-firebase/app' v8.4.7 but found v8.4.6, this might cause build issues or runtime crashes.
```

So I tried to increase my `@react-native-firebase/app` to version `8.4.7`, I then get the following error while building my app (Android). 
```
ReactNativeFirebase WARNING: NPM package '@react-native-firebase/analytics' depends on '@react-native-firebase/app' v8.4.6 but found v8.4.7, this might cause build issues or runtime crashes.
ReactNativeFirebase WARNING: NPM package '@react-native-firebase/crashlytics' depends on '@react-native-firebase/app' v8.4.6 but found v8.4.7, this might cause build issues or runtime crashes.
```

My questions is, how can I get the right peer dependency?  🤔

**Comment by mikehardy** [comment, marked as answer]
Hey @TommyLeong  -

My best advice: update to v10 of everything. Otherwise you'll have to dig through the releases and find them - it's not easy, and why we moved to a Single.True.Version in v10+

If you see that error then you probably want to update analytics and crashlytics to higher versions (you can check the release notes for each package off of rnfirebase.io/releases

On the bright side, v10 was mainly just to do that version locking. The breaking changes are effectively non-existent - just dropping already-deprecated analytics APIs that all have analogs

**Reply by TommyLeong** [reply]
Mike, if I get you right.. v10 is mainly on aligning all the RNFB modules for the peer dependency right? And of course dropping the deprecated APIs along. Rest of the features are not affected as of the latest version before v10. Did I get you right?

**Reply by mikehardy** [reply]
That is correct. There *was* a big change for ML users in the versions prior - we dropped the on-device ML features to align with google's now split between "Cloud ML" (still in Firebase) and "On Device ML" (now in Google ML Kit, soon to be wrapped in a new react-native-module by me/us). I have to mention that because it could affect you. Other than that, v10 is all about version alignment, with the only non-versioning thing being some old analytics methods that have replacements (and are noted in release notes)

**Reply by TommyLeong** [reply]
Appreciate on your details comments and confirmation! Thanks Mike 👍
